### PR TITLE
Option to specify custom delegate for NavigationDrawer menu item.

### PR DESCRIPTION
**Pull request by rmallah**
(1) This shall allow developers to use their custom delegates if the need be.

### CODE REVIEW

**Comment by rmallah** [comment]
Hi Pier , could you please have a look at this. I merged the upstream branch changes into it and 
now the travis build is passed.

File: src/imports/controls/NavigationDrawer.qml [hunk 1]
```diff
@@ -108,6 +108,13 @@ Drawer {
     */
     property list<QtObject> actions
 
+    /*!
+            \qmlproperty Component menuItemDelegate
+
+            The delegate for item that constitute a menu item.
+        */
```
Comment (plfiorini) [inline src/imports/controls/NavigationDrawer.qml]: nitpick: the documentation comment is not properly aligned

File: src/imports/controls/NavigationDrawer.qml [hunk 2]
```diff
@@ -108,6 +108,13 @@ Drawer {
...
+    property  alias menuItemDelegate : navDrawerListView.delegate
```
Comment (plfiorini) [inline src/imports/controls/NavigationDrawer.qml]: Rename `menuItemDelegate` to `delegate`

**Comment by rmallah** [comment]
thanks

**Pull request merged**